WordPress/openverse
# Fix audio snackbar and box audio seeking

## Fixes



Fixes #2742 by @obulat
Fixes #2124 by @zackkrida 

## Description


To fix the linked issues, this PR refactors the Audio search result components. 

I realized that the components that display the audio result on the All content (`VAudioCell`) and Audio search result pages (the inner part of `VAudioList`) have the same functionality. Both are `<li>` list items, and both have to handle interaction and mouse down to send analytics events. So, I refactored `VAudioCell` to `VAudioResult` that handles that, and can also set the correct audio track layout and size (`row` if it's the Audio search page, and `box` if its the All content search page). 
The new `VAudioResult` component is now used by `VAudioList` (the audio search results page and the related audio section on the single results page) and the `VAllResultsGrid` (the All content search results page).

The **snackbar handling** is moved inside the `VAudioTrack` component. This allows us to not pass as many events up anymore.

We also add `isSeekable` prop to `use-seekable` composable to prevent seeking on box audio by arrow keys.

Another change added here is to hide the related images using `brightness: 0%` and remove the `maxDiffPixels` threshold in the snapshot comparisons. Previously, there was flakiness in these snapshots because sometimes the related images did not fully load and still had gray spans when we took the snapshots. This is not related to this PR, but since this PR was started because the Playwright failures in #2735, I decided to add it here, and rebase 2735 on this PR. This will help us see the changes in the external links in that PR.

## Testing Instructions

Test that the instructions snackbar is closed when you navigate away from search results. Go to the search page, navigate to the Audio item using keyboard, and close the snackbar by pressing Spacebar to play the audio. Then, press Enter to navigate to the single result page. The snackbar should be hidden.

Test that you cannot seek on the box layout. Go to the all content search page, and navigate to an audio item using your keyboard. Press arrow keys: you shouldn't be able to "seek" the audio. You should also see no `AUDIO_INTERACTION` analytics events in the console.

Check that audio interaction and `SELECT_SEARCH_RESULT` events are sent correctly for all audio tracks. When you click on audio to navigate to the single result page, you should get the `SELECT_SEARCH_RESULT` but not `AUDIO_INTERACTION` event. When you click play or seek - you should get the `AUDIO_INTERACTION`, but no `SELECT_SEARCH_RESULT` events.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
